src/textual/widgets/_progress_bar.py

```diff
@@ -0,0 +1,432 @@
+            """
+            self.progress_bar = progress_bar
+            super().__init__()
+
+    class Started(Message):
+        """Sent when the progress bar starts tracking progress.
+
+        This message is sent when the total progress is set to a non-`None` value.
+        This message may be sent _before_ the progress bar is mounted if the progress
+        bar is instantiated with a non-`None` value for the total number of steps.
+
+        Can be handled using `on_progress_bar_started` in a subclass of
+        [`ProgressBar`][textual.widgets.ProgressBar] or in a parent widget in the DOM.
+        """
+
+        def __init__(self, progress_bar: ProgressBar, total: float) -> None:
+            """Create a message instance.
+
+            Args:
+                progress_bar: The progress bar that started tracking progress.
+                total: The total number of steps that the bar will track.
+            """
+            self.progress_bar = progress_bar
+            self.total = total
+            super().__init__()
+
+    def __init__(
+        self,
+        total: float | None = None,
+        *,
+        hide_bar: bool = False,
+        hide_percentage: bool = False,
+        hide_eta: bool = False,
+        name: str | None = None,
+        id: str | None = None,
+        classes: str | None = None,
+        disabled: bool = False,
+    ):
+        """Create a Progress Bar widget.
+
+        The progress bar uses "steps" as the measurement unit.
+
+        Example:
+            ```py
+            class MyApp(App):
+                def compose(self):
+                    yield ProgressBar(total=100)
+
+                def key_space(self):
+                    self.query_one(ProgressBar).advance()
+            ```
+
+        Args:
+            total: The total number of steps in the progress if known.
+            hide_bar: Whether to hide the bar portion from the progress bar.
+            hide_percentage: Whether to hide the percentage status from the bar.
+            hide_eta: Whether to hide the ETA countdown from the progress bar.
+            name: The name of the widget.
+            id: The ID of the widget in the DOM.
+            classes: The CSS classes for the widget.
+            disabled: Whether the widget is disabled or not.
+        """
+        super().__init__(name=name, id=id, classes=classes, disabled=disabled)
+        self._bar = Bar()
+        self._percentage_status = PercentageStatus()
+        self._eta_status = ETAStatus()
+        self.hide_bar = hide_bar
+        self.hide_percentage = hide_percentage
+        self.hide_eta = hide_eta
+
+        self._percentage = None
+        self.total = total
+
+    def compose(self) -> ComposeResult:
+        with Horizontal():
+            if not self.hide_bar:
+                bar = Bar()
+                # Notify the bar when the percentage is updated.
+                self.watch(
```

Comment:
If everything is dependant on `percentage`, can we not have a `watch_percentage` method here that updates its children?

I think that would be easier to follow than the watches and partials.